src/text/word2vec-utils.ts

```diff
@@ -0,0 +1,90 @@
+export function clean(text: string, stopwords: string[] | undefined): Array<string> {
+  let textArray = text.split(' ');
+  //filter out empty string
+  textArray = textArray.filter((value) => {
+    return value != '';
+  });
+
+  //filter out stopwords.
+  if (stopwords) {
+    textArray = textArray.filter((value) => {
+      return !stopwords.includes(value);
+    });
+  }
+  //remove punctuation
+  textArray = textArray.map((value) => {
+    return value.replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g, "");
+  });
+
+  return textArray;
+}
+
+export function skipGram(
+  window = 5, textArray: Array<string>,
+  stopwords: string[] | undefined): Array<Array<string | string[]>> {
+
+  let wordList = [];
+  let allText = [];
```

Comment:
```suggestion
  const allText = [];
```

No necessary to use let :)